### PR TITLE
chore: Move dependencies to provided scope

### LINKED ISSUE #442: Add provided scope to the vaadin dependencies
An addon is supposed to work in a Vaadin project so the library should be provided by the main project. (So maven doesn't need to omit one particular version of Vaadin).
So I think we should add the scope you can also do it on flow-client...)
```
        <dependency>
            <groupId>com.vaadin</groupId>
            <artifactId>vaadin-core</artifactId>
            <scope>provided</scope>
        </dependency>
```

Or am I mistaken?

### PR DESCRIPTION
**Pull request by caalador**
Dependencies should be provided
by the project so there are less
versioning conflicts.

Closes #442

### CODE REVIEW

**Review by samie — DISMISSED** [review]
I guess this is needed so that the resulting add-ons don't export the dependencies? Can you still develop addons as standalone project?

**Comment by caalador** [comment]
Tought it would be enough to have provided scope in jetty, but seems we need to have flow-server as a direct dependency

File: pom.xml [hunk 1]
```diff
@@ -115,13 +115,16 @@
...
             <artifactId>flow-client</artifactId>
+            <scope>provided</scope>
         </dependency>
         <dependency>
             <groupId>com.vaadin</groupId>
             <!-- Other components are probably handy when building test views -->
             <!-- Replace artifactId with vaadin to use commercial components -->
             <artifactId>vaadin-core</artifactId>
             <scope>test</scope>
+            <!-- I using vaadin dependency in addon set scope to provided -->
```
Comment (tepi) [inline pom.xml]: Should this be "If you are using vaadin..." or similar?

Comment (caalador) [reply pom.xml]: Yes should be if not I

**Review by tepi — APPROVED** [review]

**Pull request merged**